_posts/2022-03-01-auto-backport-in-opensearch.markdown

```diff
@@ -0,0 +1,46 @@
+
+4. The branches created by the backport workflow to open backport pull requests are opened against the repository. Once the pull request is merged, such branches are left over and need to be deleted manually. To resolve this problem and make the process end-to-end automated, we wanted to incorporate a feature where the source branch of the backport pull request is auto deleted after the pull request is merged. This helps to maintain branches efficiently on the repo and allows the backport workflow to clean up after itself. We have added an [auto delete](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/delete_backport_branch.yml) workflow to the release branches which can clean up the auto backport branches after the backport pull requests are merged.
+
+## Current Features
+
+The current state of auto-backport after the learnings is:
+
+1. Auto-backports are triggered by labelling a [pull request](https://github.com/opensearch-project/OpenSearch/pull/2094) with a backport label of pattern `backport <release-branch>`. Once this pull request is merged and labelled, the auto-backport workflow can run to generate a [backport pull request](https://github.com/opensearch-project/OpenSearch/pull/2106).
+2. A Github App is used to create backport pull requests so that CI workflows can trigger on such pull requests.
+3. The backport branches from such pull requests are deleted automatically once the backport pull request is merged.
+
+![Figure 1: Backport automation]({{ site.baseurl }}/assets/media/blog-images/2022-03-01-auto-backport-in-opensearch/backport-automation.png){: .img-fluid }
+
+## Closing
+
+These type of automation processes help us get to releases faster with minimum friction. Through this blog, we want to shed light on how we navigated through the waters to adopt backport automation tailored to our purpose and also share our learnings with the community. We hope this blog post helps other opensource projects in adding such automations. We would love to get your feedback and see what you think about these kinds of automations.
```

Comment:
- 'These type of automation processes help us get to releases faster with minimum friction. ' -> 'These type of automation processes help the project get to releases faster with minimum friction.'
- 'Through this blog, we want to shed light on how we navigated through the waters to adopt backport automation tailored to our purpose and also share our learnings with the community.' -> 'Through this blog, I wanted to shed some light the how the waters were navigated by the team to adopt backport automation and also share these learnings with the community.'
- 'We hope this blog post helps other opensource projects in adding such automations' -> 'I hope this blog post helps other open source projects with similar automation.'
- 'We would love to get your feedback and see what you think about these kinds of automations.' -> 'Your feedback and thoughts on this type of automation is always appreciated.'